opal/corelib/array.rb

```diff
@@ -210,32 +210,46 @@ def +(other)
     `self.concat(other)`
   end
 
-  def -(other)
-    other = if Array === other
-              other.to_a
-            else
-              `$coerce_to(other, #{Array}, 'to_ary')`.to_a
-            end
-
+  def difference(*others)
     return [] if `self.length === 0`
-    return `self.slice()` if `other.length === 0`
+
+    others.map! do |other|
+      if Array === other
+        other.to_a
+      else
+        `$coerce_to(other, #{Array}, 'to_ary')`.to_a
+      end
+    end
```

Comment:
Can you please move this into the main cycle to save an iteration?

Reply:
I'll try! 🙇